NapDatabaseExport/DatabaseProvider.cs

```diff
@@ -37,13 +37,23 @@ public virtual bool UsesUser
             get { return false; }
         }
 
+        public virtual bool RequiresUser
+        {
+            get { return UsesUser; }
+        }
+
         public string User { get; set; }
 
         public virtual bool UsesPassword
         {
             get { return false; }
         }
 
+        public virtual bool RequiresPassword
+        {
+            get { return false; }
```

Comment:
```suggestion
            get { return UsesPassword; }
```

Reply:
For consistency I think the base class should use the respective property value instead of a hard coded false.

Reply:
`RequirePassword = true` makes no sense since the original impl. Why is empty password not allowed in principle? There is no such restriction with any DB provider and this will be `false` in like 100% of the providers.

Reply:
OK